lib/cdo/firehose.rb

```diff
@@ -4,28 +4,42 @@
 require 'active_support/core_ext/module/attribute_accessors'
 
 # A wrapper client to the AWS Firehose service.
-# @example
-#   FirehoseClient.instance.put_record(
-#     {
-#       study: 'underwater basket weaving', # REQUIRED
-#       study_group: 'control',             # OPTIONAL
-#       script_id: script.id,               # OPTIONAL
-#       level_id: level.id,                 # OPTIONAL
-#       project_id: project.id,             # OPTIONAL
-#       user_id: user.id,                   # OPTIONAL
-#       event: 'drowning',                  # REQUIRED
-#       data_int: 2,                        # OPTIONAL
-#       data_float: 1.8,                    # OPTIONAL
-#       data_string: 'hello world',         # OPTIONAL
-#       data_json: "{\"key\":\"value\"}"    # OPTIONAL
-#     }
-#   )
-
-class FirehoseClient < Cdo::Buffer
-  STREAM_NAME = 'analysis-events'.freeze
-
+class FirehoseClient
   include Singleton
+
+  # Used for recording events to later be analyzed by the product team.
+  # @example
+  #   FirehoseClient.instance.put_record(
+  #     FirehoseClient::ANALYSIS_EVENTS_STREAM_NAME,
+  #     {
+  #       study: 'underwater basket weaving', # REQUIRED
+  #       study_group: 'control',             # OPTIONAL
+  #       script_id: script.id,               # OPTIONAL
+  #       level_id: level.id,                 # OPTIONAL
+  #       project_id: project.id,             # OPTIONAL
+  #       user_id: user.id,                   # OPTIONAL
+  #       event: 'drowning',                  # REQUIRED
+  #       data_int: 2,                        # OPTIONAL
+  #       data_float: 1.8,                    # OPTIONAL
+  #       data_string: 'hello world',         # OPTIONAL
+  #       data_json: "{\"key\":\"value\"}"    # OPTIONAL
+  #     }
+  #   )
+  ANALYSIS_EVENTS_STREAM_NAME = 'analysis-events'.freeze
+
+  # Stream for tracking which translated strings are used on which URLs.
+  # @example
+  #   FirehoseClient.instance.put_record(
+  #     FirehoseClient::I18N_STRING_TRACKING_EVENTS_STREAM_NAME
+  #     {
+  #       url: 'http://studio.code.org/courses',
+  #       string_key: 'home.heading_elementary_markdown'
+  #     }
+  #   )
+  I18N_STRING_TRACKING_EVENTS_STREAM_NAME = 'i18n-string-tracking-events'.freeze
```

Comment:
These names seem a bit verbose to me, I tend to prefer variables/constants passed around in business-logic code to be shorter/more readable based on context. How about something like a fixed hash to lookup the full stream name based on a short symbol, e.g.:

```ruby
STREAMS = {
  analysis: 'analysis-events',
  i18n: 'i18n-string-tracking-events'
}

def put_record(stream, data)
  stream_name = STREAMS[stream] || raise "Stream #{stream} not found"
  [...]
end

FirehoseClient.instance.put_record(:i18n,
  url: 'http://studio.code.org/courses',
  string_key: 'home.heading_elementary_markdown'
)
```